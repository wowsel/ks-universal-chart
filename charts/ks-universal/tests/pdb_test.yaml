# tests/pdb/pdb_test.yaml
suite: test pod disruption budgets
templates:
  - pdb.yaml
tests:
  - it: should create PDB with absolute minAvailable
    values:
      - ./pdb/values/absolute-numbers.yaml
    asserts:
      - matchRegex:
          path: kind
          pattern: PodDisruptionBudget
      - equal:
          path: spec.minAvailable
          value: 3

  - it: should create PDB with percentage minAvailable
    values:
      - ./pdb/values/percentage-values.yaml
    asserts:
      - matchRegex:
          path: kind
          pattern: PodDisruptionBudget
      - equal:
          path: spec.minAvailable
          value: "50%"

  - it: should create PDB with maxUnavailable
    values:
      - ./pdb/values/max-unavailable.yaml
    asserts:
      - matchRegex:
          path: kind
          pattern: PodDisruptionBudget
      - equal:
          path: spec.maxUnavailable
          value: 2