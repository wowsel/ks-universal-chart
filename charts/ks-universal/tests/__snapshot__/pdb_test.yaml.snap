should create a pod disruption budget when autoCreatePdb is enabled:
  1: |
    apiVersion: apps/v1
    kind: Deployment
    metadata:
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-test-version
      name: test-app
    spec:
      selector:
        matchLabels:
          app.kubernetes.io/component: test-app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: test-app
      template:
        metadata:
          labels:
            app.kubernetes.io/component: test-app
            app.kubernetes.io/instance: RELEASE-NAME
            app.kubernetes.io/managed-by: Helm
            app.kubernetes.io/name: test-app
            helm.sh/chart: ks-universal-test-version
        spec:
          containers:
            - image: nginx:latest
              name: main
  2: |
    apiVersion: policy/v1
    kind: PodDisruptionBudget
    metadata:
      labels:
        app.kubernetes.io/component: test-app
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: test-app
        helm.sh/chart: ks-universal-test-version
      name: test-app
    spec:
      maxUnavailable: 1
      selector:
        matchLabels:
          app.kubernetes.io/component: test-app
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: test-app
